File: gowrish/graphs/floyd_warshall/floydwarshall.cpp
#include <bits/stdc++.h>
using namespace std;

vector<vector<int>> floydwarshall(vector<vector<int>> graph)
{
    int n = graph.size();
    vector<vector<int>> dist(n, vector<int>(n, INT_MAX));

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            dist[i][j] = graph[i][j];
        }
    }

    for (int k = 0; k < n; k++)
    {
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                if (dist[i][k] != INT_MAX && dist[k][j] != INT_MAX)
                {
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j]);
                }
            }
        }
    }

    return dist;
}

int main()
{
    vector<vector<int>> graph = {
        {0, 3, INT_MAX, INT_MAX},
        {2, 0, INT_MAX, INT_MAX},
        {INT_MAX, 7, 0, 1},
        {6, INT_MAX, INT_MAX, 0}};

    vector<vector<int>> distances = floydwarshall(graph);

    cout << "Shortest distances between every pair of vertices:" << endl;
    for (const auto &row : distances)
    {
        for (int val : row)
        {
            if (val == INT_MAX)
                cout << "INF ";
            else
                cout << val << " ";
        }
        cout << endl;
    }

    return 0;
}

File: gowrish/graphs/floyd_warshall/floydwarshall.py
INF = float('inf')

def floyd_warshall(graph):
    n = len(graph)
    dist = [[INF]* n for _ in range(n)]
    
    for i in range(n):
        for j in range(n):
            dist[i][j] = graph[i][j]
    
    for k in range(n):
        for i in range(n):
            for j in range(n):
                dist[i][j] = min(dist[i][j],dist[i][k]+dist[k][j])
    return dist

graph = [
    [0, 3, INF, INF],
    [2, 0, INF, INF],
    [INF, 7, 0, 1],
    [6, INF, INF, 0]
]

distances = floyd_warshall(graph)

print("Shortest distances between every pair of vertices:")
for row in distances:
    print(row)